straxen/plugins/merged_s2s/merged_s2s.py

```diff
@@ -137,12 +161,12 @@ def compute(self, peaklets, lone_hits):
         lh = strax.sort_by_time(lh)
 
         n_top_pmts_if_digitize_top = self.n_top_pmts if "data_top" in self.dtype.names else -1
-        strax.add_lone_hits(merged_s2s, lh, self.to_pe, n_top_channels=n_top_pmts_if_digitize_top)
+        strax.add_lone_hits(merged_s2s, lh, self.to_pe, n_top_channels=n_top_pmts_if_digitize_top, waveform_start=self.save_waveform_start)
 
         strax.compute_widths(merged_s2s)
 
-        if n_top_pmts_if_digitize_top <= 0:
```

Comment:
I think you should keep logic `if n_top_pmts_if_digitize_top <= 0:` because you can not stop people form setting `sum_waveform_top_array` URLConfig to `False`.

Reply:
Is this strictly needed? I think we already cover this case with the dtype of the plugins. The `drop_data_top_field` function copies data from the input to an array with the desired dtype. If the desired dtype (the one from the plugin) does not contain the top waveform it should do it automatically without the need for the if case.

Reply:
Yes, you are right in that sense. But that will make the computing longer if we do not want to drop.

Reply:
I pushed to recover to the original implementation.